Remove deprecated error submission system from conda

degerahmet opened the pull request:
### Description

This pull request **removes the error report upload functionality** from the `conda` codebase. Issue #13900 

**Why this change was made:**
- The functionality for uploading error reports is being deprecated.
- This change simplifies error handling and removes unused or unnecessary network behavior.
- Associated tests and methods relying on the upload logic were updated accordingly.

**Changes included:**

### Removal of error report upload functionality:
- [`conda/exception_handler.py`](diffhunk://#diff-2d7d8f57f6c5a6e7fc5bed7e7c24a83b572b6b26b79a760f465a342d35368ec6L242-L382): Removed the `_upload` method and all related logic used for submitting error reports and prompting users.
- [`conda/exception_handler.py`](diffhunk://#diff-2d7d8f57f6c5a6e7fc5bed7e7c24a83b572b6b26b79a760f465a342d35368ec6L89): Updated `handle_unexpected_exception` and `handle_reportable_application_exception` to no longer invoke `_upload`.

### Updates to tests:
- [`tests/test_exceptions.py`](diffhunk://#diff-f46006e3f43ffb1dd5d6862005427f6620f4dcfb1fa2f883d8482550069eeeccL519-R521): Updated or removed test logic and assertions that validated the upload process.
  - Removed checks for HTTP requests triggered by the now-removed error report submission feature.

---

### Deprecations

* Removed error report upload functionality, including the `_upload` method and associated prompt logic in `exception_handler.py`.

### Tests

* Removed or updated tests that validated the error report submission process to reflect the removal of this functionality.

---

Just as a reminder, everyone in all Conda org spaces (including PRs) must follow the Conda Org Code of Conduct.

If you have any feedback about the contribution process, feel free to share it with us!

**Helpful links:**
- Conda Org Code of Conduct: https://github.com/conda/conda/blob/main/CODE_OF_CONDUCT.md  
- Contributing Guide: https://github.com/conda/conda/blob/main/CONTRIBUTING.md

degerahmet commented:
@conda-bot check-cla

degerahmet commented:
@conda-bot check

kenodegard commented:
@conda-bot check

degerahmet commented:
> @conda-bot check

I already signed the CLA document. Do you know when it is approved?

degerahmet commented:
@kenodegard I saw the pre-commit CI failed. I will fix it as soon as possible.

soapy1 commented on conda/exception_handler.py:
```diff
@@ -239,147 +237,6 @@ def _isatty(self):
             log.debug("%r", e)
             return True
 
-    def _upload(self, error_report) -> None:
```
Instead of removing this right away, we'll want to make this go through the deprecation cycle.

```
@deprecated(
        "26.9",
        "27.3",
        addendum="`_upload` is being deprecated and will be removed in conda version 27.3",
    )
```

soapy1 replied:
same for `_ask_upload`,`_execute_upload` and `_post_upload`

degerahmet commented:
Updated the PR: Rebased with the latest main and added @deprecated decorators as requested. Ready for review!

soapy1 commented on tests/test_exceptions.py:
```diff
@@ -753,88 +715,70 @@ def test_print_unexpected_error_message_upload_3(
 def test_print_unexpected_error_message_upload_username_with_spaces(
-    pwuid,
-    post_mock,
+    mocker: MockerFixture,
```
I think this test can be removed since we already test that an error report does not get sent in test `test_print_unexpected_error_message_upload_1`.

soapy1 reviewed (changes requested):
This is looking great, thanks again for taking a look at this. Sorry for the slow roll of feedback from me here. I think there are just a few more things to take care of before this is ready to merge:
- [x] clean up the redundant tests (as I've noted in this last round of review)
- [x] fix imports (as @jaimergp has noted)
- [x] add a news item for this change. The news item should go under the `news/` folder with a new file called something like `14698-deprecate-error-uploading-system`. You can use https://github.com/conda/conda/pull/15558/changes#diff-5ca36907170e9cf950cf1e768fafa765d4c3da7c5dfad3a7d8adfe759d2452d1 as an example. It should look something like:
```
### Enhancements

* <news item>

### Bug fixes

* <news item>

### Deprecations

* Mark `conda.exception_handler.ExceptionHandler._upload` as pending deprecation, to be removed in 27.3.  (#13900 via #14698)
* Mark `conda.exception_handler.ExceptionHandler._ask_upload` as pending deprecation, to be removed in 27.3.  (#13900 via #14698)
* Mark `conda.exception_handler.ExceptionHandler._execute_upload` as pending deprecation, to be removed in 27.3.  (#13900 via #14698)
* Mark `conda.exception_handler.ExceptionHandler._post_upload` as pending deprecation, to be removed in 27.3.  (#13900 via #14698)
* 
### Docs

* <news item>

### Other

* Disable uploading error reports.  (#13900 via #14698)
 ```

degerahmet commented:
Thanks again for the thoughtful review, really appreciate it.

I went through the items from your last round and made the updates:
- removed/condensed the redundant tests
- fixed the imports as @jaimergp 's note
- added a `news/` entry for this change (modeled after the referenced example)

All checks look good on my side. Whenever you have time, I'd love a re-review.


**Note:** I rebased the branch while applying the fixes, so older comments may have shifted slightly.

degerahmet commented:
> @degerahmet thanks, these changes are looking great! When I pull this down I see that there are still some test failures in `tests/test_exceptions.py`. It looks like this is because with the disabling of error uploading that happens in this PR we no longer call the bits that are deperecated in #15558. I have this little commit that fixes those errors up [1096842](https://github.com/conda/conda/commit/1096842acc3f79cea1be593bcd9e8e7bd438abf5)

@soapy1 Sounds good to me!

soapy1 commented on conda/exception_handler.py:
```diff
@@ -301,6 +304,11 @@ def _upload(self, error_report: dict[str, str]) -> None:
+    @deprecated(
+        "26.9",
+        "27.3",
+        addendum="`_ask_upload` is being deprecated and will be removed in conda version 27.3",
```
```suggestion
```

soapy1 commented on conda/exception_handler.py:
```diff
@@ -321,6 +329,11 @@ def _ask_upload(self) -> bool:
+    @deprecated(
+        "26.9",
+        "27.3",
+        addendum="`_execute_upload` is being deprecated and will be removed in conda version 27.3",
```
```suggestion
```

soapy1 commented on conda/exception_handler.py:
```diff
@@ -380,6 +393,11 @@ def _execute_upload(self, error_report: dict[str, Any]) -> None:
+    @deprecated(
+        "26.9",
+        "27.3",
+        addendum="`_post_upload` is being deprecated and will be removed in conda version 27.3",
```
```suggestion
```

soapy1 reviewed (approved):
Thanks again for working on this change. Just have a few knit-picks. If you don't mind, I'll go ahead and apply them right away, let the tests run, and then we should be good to merge 🚀
